test/hw3/test_task1.py

```diff
@@ -0,0 +1,18 @@
+from unittest.mock import patch
+
+from All_home_works.hw3.task1 import cache
+
+
+@cache(times=2)
+def f():
+    return input("? ")
+
+
+@patch("hw3.test_task1.f", return_value=2)
```

Comment:
Why do you need to patch function, when you can just decorate mock directly? 

mock = Mock()
function = cache(3)(mock)